Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,54 +8,43 @@ public GildedRose(Item[] items) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
 
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
+        for (Item currItem : items) {
+            switch (currItem.getItemType()) {
+                case AGED_BRIE:
+                    currItem.checkQualityAndIncrease(1);
+                    currItem.sellInDecrease();
+                    if (currItem.sellIn < 0) {
+                        currItem.checkQualityAndIncrease(1);
+                    }
+                    break;
 
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
+                case BACKSTAGE_PASSES:
+                    currItem.checkQualityAndIncrease(1);
+                    if (currItem.sellIn < 11) {
+                        if (currItem.sellIn >= 6) {
+                            currItem.checkQualityAndIncrease(1);
+                        } else {
+                            currItem.checkQualityAndIncrease(2);
                         }
                     }
-                }
-            }
+                    currItem.sellInDecrease();
+                    if (currItem.sellIn < 0) {
+                        currItem.resetQuality();
+                    }
+                    break;
 
-            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                items[i].sellIn = items[i].sellIn - 1;
-            }
+                case SULFUR:
+                    break;
 
-            if (items[i].sellIn < 0) {
-                if (!items[i].name.equals("Aged Brie")) {
-                    if (!items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].quality > 0) {
-                            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                                items[i].quality = items[i].quality - 1;
-                            }
-                        }
-                    } else {
-                        items[i].quality = items[i].quality - items[i].quality;
-                    }
-                } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
+                case DEXTERITY_VEST:
+                case MONGOOSE:
```

Comment:
consider changing that to default, or you need to add error handling e.g. throw an exception in the enum creator

Reply:
👍